invenio_users_resources/records/api.py

```diff
@@ -8,62 +8,111 @@
 # details.
 
 """API classes for user and group management in Invenio."""
-import random
+
```

Comment:
Major changes in this file and ``models.py`` to make make the record classes behave more like normal records, and to clearly separating the responsibilities of parsing the user/role/domain objects from Invenio-accounts.

Now UserAggregate/GroupAggregate/DomainAggregate are records that is responsible for harmonising data access from search index and DB. the ``<Model>AggregateModel`` objects are responsible for metadata storage and for parsing the Invenio-accounts user/role/domain models.